txtdirect.go

```diff
@@ -165,14 +165,31 @@ func contains(array []string, word string) bool {
 	return false
 }
 
-// getRecord uses the given host and path to find a TXT record
+// getRecord uses the given host to find a TXT record
 // and then parses the txt record and returns a TXTDirect record
-// struct instance It returns an error when it can't find any txt
+// struct instance. It returns an error when it can't find any txt
 // records or if the TXT record is not standard.
-func getRecord(host, path string, ctx context.Context, c Config, r *http.Request) (record, error) {
+func getRecord(host string, ctx context.Context, c Config, r *http.Request) (record, error) {
 	txts, err := query(host, ctx, c)
 	if err != nil {
-		return record{}, err
+		// if nothing found, jump into wildcards
+		hostSlice := strings.Split(host, ".")
+		hostSlice[0] = "_"
+		host = strings.Join(hostSlice, ".")
+		txts, err = query(host, ctx, c)
+		if err != nil {
+			return record{}, err
+		}
+		if txts[0] == "" {
+			// if record empty, jump into wildcards again
+			hostSlice := strings.Split(host, ".")
+			hostSlice[0] = "_"
+			host = strings.Join(hostSlice, ".")
+			txts, err = query(host, ctx, c)
+			if err != nil {
+				return record{}, err
+			}
+		}
```

Comment:
This is duplicated. First we are in the error check. That should still be separated.
And with the above code, if we have an error an empty record is implied.